### PR TITLE
Add RDS Snapshot Restore Script

### PR DESCRIPTION
**Pull request by goruha**
## What
* Script to do an in-place restore of RDS aurora cluster from snapshot with minimal downtime

## Why
* Restore databases from point-in-time backups

### CODE REVIEW

File: rds_restore_cluster_from_snapshot.sh [hunk 1]
```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/env bash
+
+set -e
+
+CLUSTER=$1
+SNAPSHOT_ID=$2
+
+MASTER_PASSWORD=$(MASTER_PASSWORD:-)
```
Comment (osterman) [inline rds_restore_cluster_from_snapshot.sh]: Wrong syntax. `${FOO:-}` (use curly braces, not `(` which means execution)

Comment (goruha) [reply rds_restore_cluster_from_snapshot.sh]: Right. 
I'm doing refactoring to reduce duplications

File: rds_restore_cluster_from_snapshot.sh [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+      PromotionTier: .PromotionTier,
+    }
+EOF
+)
+
+####==================== Restoring cluster =================================#####
+
+CLUSTER_JSON=$( aws --region $REGION rds describe-db-clusters --db-cluster-identifier "$CLUSTER" | jq -cM "$CLUSTER_MAP_QUERY" )
+
+if [ "$CLUSTER_JSON" == "" ]
```
Comment (osterman) [inline rds_restore_cluster_from_snapshot.sh]: `-z "${CLUSTER_JSON}"`

File: rds_restore_cluster_from_snapshot.sh [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+else
+  echo "WARNING: Cluster ${CLUSTER_TMP} exists"
+fi
+
+####==================== INSTANCE COUNT INFO =================================#####
+
+INSTANCES_COUNT=$( aws --region $REGION rds describe-db-instances --filter "Name=db-cluster-id,Values=${CLUSTER}" | \
+  jq -cr '.DBInstances | length' \
+)
+echo "Cluster ${CLUSTER} have ${INSTANCES_COUNT} instances"
```
Comment (osterman) [inline rds_restore_cluster_from_snapshot.sh]: s/have/has/

File: rds_restore_cluster_from_snapshot.sh [hunk 4]
```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/env bash
+
```
Comment (osterman) [inline rds_restore_cluster_from_snapshot.sh:2]: Can you describe high-level how this works at the top in comments?
e.g. 

1. Record current database settings
2. Restore snapshot to temporary name
3. Rename live database
....

File: rds_restore_cluster_from_snapshot.sh [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+      StorageType: .StorageType,
+      StorageEncrypted: .StorageEncrypted,
+      CopyTagsToSnapshot: .CopyTagsToSnapshot,
+      MonitoringInterval: .MonitoringInterval,
+      PromotionTier: .PromotionTier,
+    }
+EOF
+)
+
+####==================== Restoring cluster =================================#####
```
Comment (osterman) [inline rds_restore_cluster_from_snapshot.sh]: Can you wrap all these sections in functions and use comments like

```
#
# Restore cluster from snapshot
#
```

**Review by osterman — APPROVED** [review]
Nice refactoring

**Review by osterman — CHANGES_REQUESTED** [review]
s/not exists/does not exist/

**Review by osterman — APPROVED** [review]

**Pull request merged**